base_classes/NXobject.nxdl.xml

```diff
@@ -30,5 +30,164 @@
         <!--attribute name="name"><doc>name of instance</doc></attribute-->
-</definition>
+	<field name="identifierNAME" type="NX_CHAR" nameType="partial">
+		<doc>
+			An identifier for a (persistent) resource.
 
+			An identifier, provided by some authority, that has been assigned to the
+			real-world object described by this ``NXobject``. To be useful, the identifier
+			must not be reassigned to a different real-world object. It is typical for
+			there to be some mechanism to resolve an identifier, obtaining metadata
+			about the object. Identifiers for which some guarantees exist regarding
+			this resolution process are called persistent identifiers.
+			Persistent identifiers are also known as PIDs.
+		</doc>
+		<attribute name="type" type="NX_CHAR">
+			<doc>The type of identifier used.</doc>
+			<enumeration>
+				<item value="ARK">
+					<doc>
+						Archival Resource Key.
+						
+						An ARK is a Uniform Resource Identifier (URI) designed to support long-term access to a variety of information objects.
+						
+						Syntax: https://NMA/ark:/NAAN/Name[Qualifier]. Brackets indicate optional elements.
+						Example: https://example.org/ark:/12345/abcde
+					</doc>
+				</item>
+				<item value="DOI">
+					<doc>
+						Digital Object Identifier.
+						
+						A DOI is a unique alphanumeric string used to identify digital content. It consists of a prefix and a suffix, separated by a slash.
+						
+						Syntax: 10.XXXX/XXXXXX
+						Example: 10.1107/S1600576714027575
+					</doc>
+				</item>
+				<item value="Hdl">
+					<doc>
+						A handle is a unique identifier that consists of a prefix indicating the naming authority and a suffix representing the local name 
+						of a resource, cts. 
+						
+						A handle is a unique identifier used to facilitate the identification and management of digital objects. It is composed of a prefix that 
+						indicates the naming authority and a suffix that specifies the resource's local name. 
+						
+						This refers specifically to an ID in the Handle system operated by the Corporation for National Research Initiatives (CNRI).
+						
+						Syntax: hdl:prefix/identifier
+						Example: hdl:123456789/abc123
+					</doc>
+				</item>
+				<item value="IGSN">
+					<doc>
+						International Generic Sample Number.
+						
+						The IGSN is a unique identifier assigned to a specific sample or specimen in the context of scientific research.
+						
+						Syntax: https://igsn.org/{IGSN-ID}
+						Example: https://igsn.org/IGSN:XYZ123456
+					</doc>
+				</item>
+				<item value="ISNI">
+					<doc>
+						ISNI is an ISO standard to uniquely identify individuals and organizations involved in creative work, including pseudonyms
+						and other public personas.
+						
+						An ISNI is made up of 16 digits, the last character being a check character. The check character may be either a decimal digit
+						or the character “X”. 
+						
+						Syntax: https://isni.org/isni/{ISNI-ID}
```

Comment:
Do we really need the `https://isni.org/isni/` prefix?

Couldn't we store the ISNI-ID number and document that a URL may be derived from the ISNI by adding the prefix?

i.e., 
    Syntax: 16 base-10 digits stored without any spaces.

Apparently, specifying the lack of spaces [is important](https://en.wikipedia.org/wiki/International_Standard_Name_Identifier#ISNI_format).

Reply:
Sounds good. I used
```
A URL can be generated from the ISNI ID by combining it with the prefix https://isni.org/isni/, resulting in
https://isni.org/isni/{ISNI-ID}.
						
Syntax: 16 base-10 digits stored without any spaces.
Example: 0000000121032683
```